### PR TITLE
Remove `kv_database_config` from snapshots starting in Spring 1.0 (snapshot V7)

### PR DESCRIPTION
**Pull request by linh2931**
`kv_database_config` has been stored in snapshot's `global_property_object` since `V4` but it has never been used.
* Spring 1.0 and future releases stop storing it in snapshot.
* Snapshots V4/V5/V6 still keep it for backward compatibilities.
* `kv_config.hpp` has been updated to warn against future uses.

Resolves https://github.com/AntelopeIO/spring/issues/102

### CODE REVIEW

File: libraries/chain/controller.cpp [hunk 1]
```diff
@@ -2216,8 +2217,7 @@ struct controller_impl {
                   section.read_row(legacy_global_properties, db);
 
                   db.create<global_property_object>([&legacy_global_properties,&gs_chain_id](auto& gpo ){
-                     gpo.initialize_from(legacy_global_properties, gs_chain_id, kv_database_config{},
-                                        genesis_state::default_initial_wasm_configuration);
+                     gpo.initialize_from(legacy_global_properties, gs_chain_id);
```
Comment (heifner) [inline libraries/chain/controller.cpp:2220]: What about genesis_state::default_initial_wasm_configuration ?

Comment (linh2931) [reply libraries/chain/controller.cpp:2220]: It is now filled in by `initialize_from()` for simplication.

Comment (heifner) [reply libraries/chain/controller.cpp:2220]: Thanks. Hard to see on phone. I'll review tomorrow.

Comment (linh2931) [reply libraries/chain/controller.cpp:2220]: Thank you and appreciate it! You don't need to review the PR on your vacation. I put it up for making sure all CICD pass before you and Greg look it next Monday.

File: libraries/chain/include/eosio/chain/kv_config.hpp [hunk 2]
```diff
@@ -12,8 +12,10 @@ namespace eosio { namespace chain {
     * The key and value limits apply when adding or modifying elements.  They may be reduced
     * below existing database entries.
     * 
-    * This has been marked for removal at a later date, when v7 snapshots are deemed necessary.
-    * kv_database_config will be removed as part of that effort.
+    * Current uses of kv_database_config were removed in Spring 1.0.
+    * Do not use it in the future.
+    * The file is kept for backward compatibilities as it was used in snapshots
+    * containing snapshot_global_property_object_v4 and snapshot_global_property_object_v4.
```
Comment (heifner) [inline libraries/chain/include/eosio/chain/kv_config.hpp:18]: Fix comment

**Review by heifner — APPROVED** [review]

**Review by greg7mdp — APPROVED** [review]

**Comment by ericpassmore** [comment]
Note:start
group: IF
category: INTERNALS
summary: Remove unused key value database config starting from v7 snapshots. 
Note:end

**Pull request merged**